src/components/datepicker/datePicker.js

```diff
@@ -221,6 +279,9 @@
     /** @type {boolean} Whether the date-picker's calendar pane is open. */
     this.isCalendarOpen = false;
 
+    /** @type {boolean} Whether or not the datepicker is being used as a date-range picker. */
+    this.isDateRangePicker = null;
```

Comment:
I don't think $element[0]/tagName would help as this value ("MD-DATEPICKER") would always be the same regardless of the context under which the the md-datepicker is being used (remeber we are not creating a separate directive). I can however use something like this --> $element[0].hasAttribute('ng-model'). So if that attribute is missing, I can assume that the date-range-picker context is being used. Great suggestion!